src/messages.rs

```diff
@@ -487,13 +489,11 @@ where
         Err(_) => return Err(Error::SocketError),
     };
 
-    let mut bytes = BytesMut::with_capacity(len as usize + 1);
-
-    bytes.put_u8(code);
-    bytes.put_i32(len);
-    bytes.put_slice(&buf);
+    buffer.put_u8(code);
+    buffer.put_i32(len);
+    buffer.put_slice(&buf);
```

Comment:
```suggestion
    buffer.put_slice(&buffer);
```
i think this would allow us to get rid of the `let mut buf` allocation, cc @levkk correct me.

Reply:
Removing the creation of this buffer to read the message body was a huge lift! Great idea :D